### ISSUE TITLE
New `Style/HashEachMethods` false positive with v1.59 after "false positives fixes"

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop/rubocop#12534]
## Expected behavior

No offenses, probably.

Or should I act in another way?

There is just an array of arrays. And specs are failing without deep modifications.

## Actual behavior

There is an offense from `Style/HashEachMethods` cop with RuboCop 1.59.0.

```
Offenses:

lib/r18n-core/i18n.rb:203:9: C: [Correctable] Style/HashEachMethods: Use each_value instead of each and remove the unused _place block argument.
        available_in_places.each do |_place, available| ...
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

91 files inspected, 1 offense detected, 1 offense autocorrectable
```

## Steps to reproduce the problem

Here is a PR: https://github.com/r18n/r18n-core/pull/99

Here is a part of code:

```ruby
available_in_places = @translation_places.map { |i| [i, i.available] }

# ...

unless defined? @locale
  available_in_places.each do |_place, available|
    # ...
  end
end
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-performance 1.19.1
  - rubocop-rake 0.6.0
```

**Comment by koic** [comment]
The `Style/HashEachMethods` cop has already been marked as unsafe. This means that its autocorrection is also unsafe.

> This cop is unsafe because it cannot be guaranteed that the receiver is a `Hash`. The `AllowedReceivers` configuration can mitigate, but not fully resolve, this safety issue.

https://docs.rubocop.org/rubocop/1.59/cops_style.html#stylehasheachmethods

In this case, in addition to the auto-correction, you will need to modify the code as follows:

```diff
diff --git a/lib/r18n-core/i18n.rb b/lib/r18n-core/i18n.rb
index 8f762fe..187b678 100644
--- a/lib/r18n-core/i18n.rb
+++ b/lib/r18n-core/i18n.rb
@@ -197,7 +197,7 @@ module R18n

       available_in_places = @translation_places.map { |i| [i, i.available] }
       available_in_extensions =
-        R18n.extension_places.map { |i| [i, i.available] }
+        R18n.extension_places.to_h { |i| [i, i.available] }

       unless defined? @locale
         available_in_places.each do |_place, available|
```

**Comment by AlexWayfer** [comment]
> In this case, in addition to the autocorrection, the code will need to be manually modified as follows:

OK, thanks. I've noticed that these variables are local with a few interactions, so it's safe to convert them into a Hash.



> The `Style/HashEachMethods` cop has already been marked as unsafe. This means that its autocorrection is also unsafe.

I'll close this issue, but probably "unsafe" mark for cops should not only affect auto-correction, but also enabling in `AllCops: NewCops`. Or, at least, has some notice in RuboCop's output near offense, like "yes, there is an offense, but this cop is unsafe and non-relatable (probably the reason), so be careful".

**Comment by mishina2228** [comment]
I know this issue has already been closed, but I hit a similar case and will mention it.

```ruby
{ b: 1, a: 2 }.sort.each do |_key, value|
  p value
end
```

```
Offenses:

lib/foo.rb:3:1: C: [Correctable] Style/HashEachMethods: Use each_value instead of each and remove the unused _key block argument.
{ b: 1, a: 2 }.sort.each do |_key, value| ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

According to the doc, enclosing block arguments in parentheses suppressed RuboCop offenses.
```ruby
{ b: 1, a: 2 }.sort.each do |(_key, value)|
  p value
end
```
